packages/base/src/commands.ts

```diff
@@ -416,4 +487,110 @@ namespace Private {
       await dialog.launch();
     };
   }
+
+  export async function getUserInputForRename(
+    text: HTMLElement,
+    input: HTMLInputElement,
+    original: string
+  ): Promise<string> {
+    const parent = text.parentElement as HTMLElement;
+    parent.replaceChild(input, text);
+    input.focus();
+
+    return new Promise<string>(resolve => {
+      input.addEventListener('blur', () => {
+        parent.replaceChild(text, input);
+        resolve(input.value);
+      });
+
+      input.addEventListener('keydown', (event: KeyboardEvent) => {
+        if (event.key === 'Enter') {
+          event.stopPropagation();
+          event.preventDefault();
+          input.blur();
+        } else if (event.key === 'Escape') {
+          event.stopPropagation();
+          event.preventDefault();
+          input.value = original;
+          input.blur();
+          text.focus();
+        }
+      });
+    });
+  }
+
+  export function removeSelectedItems(
+    model: IJupyterGISModel | undefined,
+    itemTypeToRemove: SelectionType,
+    removeFunction: (id: string) => void
+  ) {
+    const selected = model?.localState?.selected.value;
+
+    if (!selected) {
+      console.info('Nothing selected');
+      return;
+    }
+
+    for (const selection in selected) {
+      if (selected[selection].type === itemTypeToRemove) {
+        removeFunction(selection);
+      }
+    }
+  }
+
+  export async function renameSelectedItem(
+    model: IJupyterGISModel | undefined,
+    itemType: SelectionType,
+    callback: (itemId: string, newName: string) => void
+  ) {
+    const selectedItems = model?.localState?.selected.value;
+
+    if (!selectedItems) {
+      console.error(`No ${itemType} selected`);
+      return;
+    }
+
+    let itemId = '';
+
+    // If more then one item is selected, only rename the first
+    for (const id in selectedItems) {
+      if (selectedItems[id].type === itemType) {
+        itemId = id;
+        break;
+      }
+    }
+
+    if (!itemId) {
+      return;
+    }
+
+    const nodeId = selectedItems[itemId].selectedNodeId;
+    if (!nodeId) {
+      return;
+    }
+
+    const node = document.getElementById(nodeId);
+    if (!node) {
+      console.warn(`Node with ID ${nodeId} not found`);
+      return;
+    }
+
+    const edit = document.createElement('input');
+    edit.classList.add('jp-gis-left-panel-input');
+    const originalName = node.innerText;
```

Comment:
We could add the `originalName` as input value, to avoid typing everything again for a small change